### PR TITLE
Bugfix builder dialog: error when changing values in table

### LINKED ISSUE #257: Builder: Error message when changing values in table
```
QAbstractItemView::commitData called with an editor that does not belong to this view
QAbstractItemView::closeEditor called with an editor that does not belong to this view
```

### PR DESCRIPTION
**Pull request by adrianusler**
I have actually just restructured the code and I am not sure why this solved the problem, but it seems like it did.
Closes #257.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced atom addition process and error handling in the molecule builder.
	- Improved efficiency in updating the z-matrix and calculating positions for new atoms.

### CODE REVIEW

File: src/molara/Gui/builder.py [hunk 1]
```diff
@@ -89,29 +90,20 @@
...
-                self.add_second_atom(mol, params)
-            elif mol.n_at == 2:  # noqa: PLR2004
-                self.add_third_atom(mol, params, atom_nums)
-            else:
-                self.add_nth_atom(mol, params, atom_nums)
+            num_atoms = self.main_window.mols.mols[0].n_at
+        params, atom_ids = self._get_parameters(num_atoms)
+
+        self.add_atom(num_atoms, params, atom_ids)
+        mol = self.main_window.mols.mols[0]
```
[the remark this answers is not in the document]
Comment (ab5424) [reply src/molara/Gui/builder.py:100]: Yes.

**Comment by ab5424** [comment]
@adrianusler Will you add some tests for the builder or should we merge it as-is?

**Comment by adrianusler** [comment]
@ab5424 I'm not sure. Tests would be nice, but it's generally better not to wait to long with the merges...

**Comment by Michel-Heinz** [comment]
Can you merge the main branch into this?

**Comment by adrianusler** [comment]
Ready to merge @Michel-Heinz!

**Review by Michel-Heinz — APPROVED** [review]

**Pull request merged**